primitives/arithmetic/src/per_things.rs

```diff
@@ -496,11 +496,34 @@ where
 	(x / maximum) * part_n + c
 }
 
+/// Checked compute of the error due to integer division in the expression `x / denom * numer`.
+fn checked_rational_mul_correction<N, P>(
+	x: N,
+	numer: P::Inner,
+	denom: P::Inner,
+	rounding: Rounding,
+) -> Result<N, &'static str>
+where
+	N: MultiplyArg + UniqueSaturatedInto<P::Inner>,
+	P: PerThing,
+	P::Inner: Into<N>,
+{
+	if P::Inner::is_zero(&denom) {
+		return Err("Division by zero")
+	}
+	Ok(rational_mul_correction::<N, P>(x, numer, denom, rounding))
```

Comment:
Now you have the implication that the underlying function can only error iff `denom` is null, which is an implementation detail. Again i would rather move the logic into the checked variant and then call it from the unchecked one.